src/app/[locale]/dashboard/quizes/_components/quiz-card/components/quiz-dialog.tsx

```
@@ -30,12 +30,18 @@ export default async function QuizFormDialog({ quiz }: QuizFormDialogProps) {
           <DialogDescription />
         </DialogHeader>
 
-        {questions.questions.length > 0 ? (
-          <QuizForm quiz={quiz} questions={questions.questions} />
+        {!payload.success ? (
+          <p className="text-center text-lg text-red-500">{payload.error}</p>
```

You can follow this approach by returning a different element if an error occures, or you can throw the error BUT you will have to catch it in the nearest `error.tsx` file which will replace this entire route, so do we catch it here to throw it to catch it later in the `error.tsx` file? It's up to you, but in this case. catching it here is better for UX.